polkadot/xcm/src/lib.rs

```diff
@@ -22,6 +22,9 @@
 // Hence, `no_std` rather than sp-runtime.
 #![cfg_attr(not(feature = "std"), no_std)]
 
+// Because of XCMv2.
+#![allow(deprecated)]
```

Comment:
@franciscoaguirre why not just remove V2 when V5? I think we should maybe start deprecating when adding new one.
E.g let's say we have v2 v3 and when adding v4, we just mark v2 as deprecated, so when adding v5 we just remove deprecated v2 and so on.

Reply:
I agree with the idea. That was the missing part. We added v4 but didn't deprecate v2. So now we deprecate v2 and remove it when we have v5.

Reply:
> I agree with the idea. That was the missing part. We added v4 but didn't deprecate v2. So now we deprecate v2 and remove it when we have v5.

exactly, ok, very cool, we are on the same boat :)

To my other comments about - reverting removed tests from XCM modules, marking `#[deprecated]` other types... My point here is to make this PR as a very good example for deprecating future versions and I would go with minimal modifications - e.g. just to add `#[deprecated]` with note "will be removed when adding new v5", I would not mention June, but also it is ok.

E.g. isn't it enough just to deprecated the whole `pub mod v2;` [here](https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/xcm/src/lib.rs#L31) instead of deprecating particular types?

Also, we use `xcm::v2` outside of `xcm` directory, so yes, there I would maybe remove or rewrite those tests to use xcm::v3/v4.

Reply:
Yeah I could not mention June. I was just following the [deprecation checklist](https://github.com/paritytech/polkadot-sdk/blob/master/docs/contributor/DEPRECATION_CHECKLIST.md) suggestions.
I can just mention v5.